core/sql/generator/GenPreCode.cpp

```diff
@@ -4368,14 +4368,6 @@ RelExpr * GenericUpdate::preCodeGen(Generator * generator,
 	  identityCol = valId.getNAColumn();
 	}
 
-      if ((getOperatorType() != REL_HBASE_UPDATE) &&
-	  (mergeInsertRecExpr().entries() > 0))
-	{
-	  *CmpCommon::diags() << DgSqlCode(-3241)
-			      << DgString0(" Non-unique ON clause not allowed with INSERT.");
-	  GenExit();
-	}
-
```

Comment:
I am glad that this check was removed. Can you please tell me if we ever see any of the other 3241 errors being triggered from the generator? To me this indicates that the binder "let one get through". I would like to eventually raise all these 3241 errors in thebinder.